vmdb/spec/models/ems_refresh/refreshers/openstack_refresher_rhos_havana_errors_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+
+describe EmsRefresh::Refreshers::OpenstackRefresher do
+  before(:each) do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    @ems = FactoryGirl.create(
+      :ems_openstack,
+      :zone      => zone,
+      :hostname  => "1.2.3.4",
+      :ipaddress => "1.2.3.4",
+      :port      => 5000)
+    @ems.update_authentication(:default => {:userid => "admin", :password => "password"})
+  end
+
+  it "will record an error when trying to perform a full refresh against RHOS Havana" do
+    refresh_ems_with_cassette(@ems, "#{described_class.name.underscore}_rhos_havana_with_error")
+
+    assert_failed_refresh
+  end
+
+  def assert_failed_refresh
+    @ems.refresh_status.should == "FAILED"
+  end
+
+  def refresh_ems_with_cassette(ems, cassette_name)
```

Comment:
Can you clarify that you modified the casette and how you modified it to induce the error?